asio/cmake/WarningsAsErrors.cmake

```diff
@@ -0,0 +1,12 @@
+if(CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND CMAKE_CXX_SIMULATE_ID MATCHES "MSVC")
```

Comment:
This code should be in the user presets file. Definitely not hardcoded in the CMake code.

Warnings as errors is also counterproductive:  
https://youtu.be/bsXLMQ6WgIk?t=3912  
https://youtu.be/bXx3-drA_oI

Reply:
yes, but are  `PRIVATE`, not exported.

Reply:
It's still a lot of code for something that can be a short, single line in the presets. Developer mode is also erroneously on by default, which would affect builds when it shouldn't.

Reply:
not longer need